src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -189,12 +196,22 @@ def _process_event_queue(self):
 
         self._event_queue.task_done()
 
+    def _check_invalidation_watcher_liveness(self):
+        time.sleep(self.INVALIDATION_WATCHER_LIVENESS_CHECK_INTERVAL)
```

Comment:
Is this safe for things like flakiness? I've become skeptical of `time.sleep`, but I also don't have any suggestions for alternatives so only want to raise it as a possible issue.

Reply:
Yea, this is fine. All of these services are running in infinite loops with small sleeps.